geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonPathParser.scala

```diff
@@ -128,6 +132,10 @@ private class JsonPathParser extends Parser {
 
   def Attribute: Rule1[PathAttribute] = rule { "." ~ oneOrMore(Character) ~> PathAttribute ~ !"()" }
 
+  def BracketedAttribute: Rule1[BracketedPathAttribute] = rule { ".['" ~ oneOrMore(CharacterWithDelimiter) ~> BracketedPathAttribute ~ !"()" ~ "']" }
+
+  def BracketedAttribute2: Rule1[BracketedPathAttribute] = rule { ".[" ~ oneOrMore(CharacterWithDelimiter) ~> BracketedPathAttribute ~ !"()" ~ "]" }
```

Comment:
I went down this route since GeoTools Literals might not be able to contain single quotes...  

The consequence is that we are getting a GeoMesa-specific, JsonPath-like syntax.